tests/lib/external/snapd-testing-tools/utils/log-filter

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+
+import argparse
+import io
+import re
+import sys
+
+import log_helper as helper
+
+# Rules are applied to filter content in the following operation details:
+# ERROR: The error shoes the bash output when a task fails
+# DEBUG: The debug includes the output of the debug script for task/suite errors
+# WARNING: The last lines from the task execution output
+# FAILED: Include a list of failed tests in the results section
+SUPPORTED_RULES = [helper.Result.FAILED.value] + helper.ExecutionInfo.list()
+
+
+def print_line(line: str):
+    if not line:
+        print()
+    else:
+        print(line.strip())
+
+
+def write_line(line: str, to_stream: io.TextIOWrapper):
```

Comment:
@zyga wrote:
Please combine print_line and write_line, we have print(..) optional argument to do both.